src/surge-xt/gui/overlays/LuaEditors.h

```diff
@@ -88,32 +88,77 @@ struct FormulaModulatorEditor : public CodeEditorContainerWithApply, public Refr
                            Surge::GUI::Skin::ptr_t sk);
     ~FormulaModulatorEditor();
 
-    std::unique_ptr<ExpandingFormulaDebugger> debugPanel;
-    std::unique_ptr<FormulaControlArea> controlArea;
     void resized() override;
+    void onSkinChanged() override;
     void applyCode() override;
-
+    void forceRefresh() override;
+    void setApplyEnabled(bool b) override;
     void showModulatorCode();
     void showPreludeCode();
-
     void escapeKeyPressed();
 
     void updateDebuggerIfNeeded();
 
+    std::unique_ptr<juce::CodeDocument> preludeDocument;
+    std::unique_ptr<juce::CodeEditorComponent> preludeDisplay;
+    std::unique_ptr<FormulaControlArea> controlArea;
+
+    std::unique_ptr<ExpandingFormulaDebugger> debugPanel;
+
     LFOStorage *lfos{nullptr};
     FormulaModulatorStorage *formulastorage{nullptr};
     SurgeGUIEditor *editor{nullptr};
     int lfo_id, scene;
+    int32_t updateDebuggerCounter{0};
 
-    void onSkinChanged() override;
-    void setApplyEnabled(bool b) override;
+    DAWExtraStateStorage::EditorState::FormulaEditState &getEditState();
+
+    bool shouldRepaintOnParamChange(const SurgePatch &patch, Parameter *p) override
+    {
+        return false;
```

Comment:
This is fine for now but we should add an item to a list which is like "what happens if the wt editor is torn out and you change oscillator from wavetable to twist" or some such. We should make that a conscious decision at least.

Reply:
Did a little test and I think this works fine, changing oscillator type does not affect the torn out window and the WT generation still works, ie. the generated wavetable is ready to go after switching back to the WT oscillator.